runtime/lua/vim/treesitter/query.lua

```diff
@@ -590,18 +639,27 @@ function Query:match_preds(match, pattern, source)
     -- continue on the other case. This way unknown predicates will not be considered,
     -- which allows some testing and easier user extensibility (#12173).
     -- Also, tree-sitter strips the leading # from predicates for us.
-    local pred_name ---@type string
-
-    local is_not ---@type boolean
+    local is_not = false
+    local is_any = false
 
     -- Skip over directives... they will get processed after all the predicates.
     if not is_directive(pred[1]) then
-      if string.sub(pred[1], 1, 4) == 'not-' then
-        pred_name = string.sub(pred[1], 5)
-        is_not = true
-      else
-        pred_name = pred[1]
-        is_not = false
+      local pred_name = pred[1] --[[@as string]]
+      while true do
+        if pred_name:match('^not%-') then
+          pred_name = pred_name:sub(5)
+          is_not = true
+        elseif pred_name:match('^any%-') then
```

Comment:
Related question: Do we want to make `any-` a general prefix as `not-` does? In other words, would we also allow predicates like `any-has-parent?` `any-nth?` in general?

We can consider applying and allowing `any-` to certain predicates only, e.g., `any-eq?` `any-match?` `any-not-match` `any-of`.

Reply:
So any-of? -> of? :D 

I'm not against it

Reply:
Due to the colliding `any-of` was handled as an exception. My suggestion is that we don't add `any-` as a general *quantifier*.

Reply:
My suggestion is not to deviate from upstream. We worked hard to align _more_; this would be a (massive) step backward.